include/zarray/zassign.hpp

```diff
@@ -15,6 +15,32 @@
 
 namespace xt
 {
+    namespace detail
+    {
+        template <class Tag>
+        struct zexpression_assigner
+        {
+            // Both E1 and E2 are zarray expressions
+            template <class E1, class E2>
+            static void assign_data(E1& e1, const E2& e2)
+            {
+                e2.assign_to(e1.get_implementation());
+            }
+        };
+
+        template <>
+        struct zexpression_assigner<xtensor_expression_tag>
+        {
+            // E1 is a zarray_expression, E2 is an xtensor_expression
+            template <class E1, class E2>
+            static void assign_data(E1& e1, const E2& e2)
+            {
+                using value_type = typename E2::value_type;
+                e1.template get_array<value_type>() = e2;
```

Comment:
What if the array wrapped by `e1` is not of type `E2::value_type`?

Reply:
I'm afraid it won't work

Reply:
What should be the desired behavior? Should it cast the source type to the destination type, or create a new array with the source data type?

Reply:
Actually I wonder if the assign operator should not xwrap the xtensor expression in a temporary zarray before dispatching the assignment. But this would not work with `noalias`.

Creating a new array with the correct `value_type` could be a solution, but that would be dangerous because you may accidentally erase an array when you want to read while it is not const:

```cpp
xt::xarray<double> a;
zarray<double> za(a);
auto c = za.get_array<int>(); // Dangerous: za now wraps an empty array of int.
std::cout << c(0, 0) << std::endl;
```